apps/ejabberd/src/ejabberd_c2s.erl

```diff
@@ -810,11 +812,17 @@ do_open_session(El, JID, StateData) ->
     ?INFO_MSG("(~w) Opened session for ~s", [StateData#state.socket, jid:to_binary(JID)]),
     Res = jlib:make_result_iq_reply(El),
     Packet = {jid:to_bare(StateData#state.jid), StateData#state.jid, Res},
-    case send_and_maybe_buffer_stanza(Packet, StateData, wait_for_session_or_sm) of
-        {_, _, NStateData, _} ->
+    Acc = mongoose_acc:new(),
```

Comment:
From what I see function `do_open_session` is called only from `maybe_open_session`. This one is only called from `wait_for_session_or_sm`. Would it be worth to create the acc (from the stanza user sent) before `maybe_open_session` is called and pass as an argument?